aten/src/ATen/native/quantized/QTensor.cpp

```diff
@@ -100,17 +100,17 @@ Quantizer* quantizer(const Tensor& self) {
 
 Tensor int_repr_quant(const Tensor& self) {
   Tensor dst;
-  // TODO: replace with TensorIterator
-  auto self_c = self.contiguous();
   AT_DISPATCH_QINT_TYPES(self.scalar_type(), "int_repr", [&]() {
     dst = at::empty(self.sizes(), self.options().dtype(UNDERLYING_TYPE));
     underlying_t* self_data =
-        reinterpret_cast<underlying_t*>(self_c.data_ptr<scalar_t>());
+        reinterpret_cast<underlying_t*>(self.data_ptr<scalar_t>());
     underlying_t* dst_data = dst.data_ptr<underlying_t>();
     if (self.numel() > 0) {
       memcpy(dst_data, self_data, self.nbytes());
```

Comment:
wait a sec, you can't just memcpy if the original tensor is not contiguous. You'd need to use one of the fancy copy functions (probably pull out from cpu implementation of copy_)

Reply:
looks like fancy copy uses TensorIterator. I actually started with TensorIterator, but had some problem with type promotion. Problem is looks like TensorIterator will always compute common type of input and output? https://bddppq.github.io/codebrowser/pytorch/pytorch/aten/src/ATen/native/TensorIterator.cpp.html#125 but what I need is a function from QUInt8 -> UInt8.

Reply:
Hm, not sure. @VitalyFedyunin @li-roy might have a better idea

Reply:
Indeed you can't memcpy if your tensor is not contiguous. And indeed TensorIterator currently have problems with in/out types which is @ifedan aiming to solve soon-ish.

Reply:
ok, let's keep it contiguous+memcpy for now then?

Reply:
I'm currently working on PR for TensorIterator, that will allow to compute and propagate common type only from inputs, so output will be able to have any type. Will it solve your problem?

Reply:
@ifedan I think so, please let me know when you land the PR

Reply:
@dzhulgakov we need this change to test view/reshape for quantized tensor, because we need to preserve the memory format when we call int_repr in order to know the memory format are changed properly

Reply:
There is PR for type promotion: https://github.com/pytorch/pytorch/pull/22273. I will wait until it will be landing to continue working on comparison ops.